libs/core/src/lib/createMock.ts

```diff
@@ -68,43 +66,42 @@ export const createMock = <T extends Options = Options>(
     .getState()
     .mocks.find((stateData) => stateData.mockTitle === mockTitle);
 
+  const createMockBaseArgs = {
+    mockFn,
+    openPageURL,
+  };
+
   let convertedConfig = convertMockOptions(
     initialState?.mockOptions || mockOptions
   );
 
   const returnValue: CreateMockFnReturnType<T> = {
-    mocks: setupMocks(convertedConfig, mockFn),
+    mocks: initializeMocks(convertedConfig, mockFn),
     mockTitle,
-    mockOptions: initialState?.mockOptions || mockOptions,
-    updateMock: (updateValues, updateScenarioMock) => {
-      if (updateScenarioMock) {
-        return {
-          ...returnValue,
-          mockOptions: updateMockOptions(mockOptions, updateValues, true),
-          mocks: setupMocks({ ...convertedConfig, ...updateValues }, mockFn),
-        };
-      } else {
-        convertedConfig = { ...convertedConfig, ...updateValues };
-        returnValue.mocks = setupMocks(convertedConfig, mockFn);
-        state.updateMock({
-          mockTitle,
-          mockOptions: updateMockOptions(mockOptions, updateValues),
-          openPageURL: getPageURL(convertedConfig, openPageURL),
-          updateMock: returnValue.updateMock,
-          resetMock: returnValue.resetMock,
-        });
-      }
+    updateMock: (updateValues) => {
+      convertedConfig = { ...convertedConfig, ...updateValues };
+      // TODO: is mutating an returned object an anti pattern? Something tells me it might be buggy
```

Comment:
// TODO: investigate race patterns here. Imo it adds complexity. there must be a better way for state management.